src/diff/mount.js

```diff
@@ -199,6 +200,9 @@ function mountDOMElement(dom, internal, globalContext, isSvg, commitQueue) {
 			);
 		}
 
+		// To override mountChildren setting _dom
+		internal._dom = dom;
+
```

Comment:
Why do we set `internal._dom = dom` so many times in this function you ask? Well currently `diffChildren` sets `parentInternal._dom = firstDomChild` after all children are diffed so this line overrides that change. Once we remove the `_dom` property on Components, all that extra dom tracking (including the `parentInternal._dom = firstDomChild` line) will go away.

Reply:
Actually, as I type this I'm wondering if instead we should just wrap the `parentInternal._dom = firstChildDom` in an if statement that only applies it for TYPE_COMPONENT... I may do that instead.

Reply:
Done